app/javascript/components/explore/ExploreDisplayTabs.jsx

```diff
@@ -371,87 +385,152 @@ export default function ExploreDisplayTabs({
               <ExploreConsensusSelector
                 consensus={exploreParamsWithDefaults.consensus}
                 updateConsensus={consensus => updateExploreParams({ consensus })}/>
-            }
-            { !!exploreInfo?.inferCNVIdeogramFiles &&
+              }
+              { !!exploreInfo?.inferCNVIdeogramFiles &&
                 <InferCNVIdeogramSelector
                   inferCNVIdeogramFile={exploreParamsWithDefaults.ideogramFileId}
                   studyInferCNVIdeogramFiles={exploreInfo.inferCNVIdeogramFiles}
                   updateInferCNVIdeogramFile={updateInferCNVIdeogramFile}
                 />
+              }
+            </div>
+            <PlotDisplayControls
+              shownTab={shownTab}
+              exploreParams={exploreParamsWithDefaults}
+              updateExploreParams={updateExploreParams}
+              allGenes={exploreInfo ? exploreInfo.uniqueGenes : []}/>
+            {hasDE &&
+            <>
+              <button
+                className="btn btn-primary"
+                onClick={() => {setShowDeGroupPicker(true)}}
+              >Differential expression</button>
+              <br/><br/>
+            </>
             }
-          </div>
-          <PlotDisplayControls
-            shownTab={shownTab}
-            exploreParams={exploreParamsWithDefaults}
-            updateExploreParams={updateExploreParams}
-            allGenes={exploreInfo ? exploreInfo.uniqueGenes : []}/>
-          <button className="action"
-            onClick={clearExploreParams}
-            title="reset all view options"
-            data-analytics-name="explore-view-options-reset">
-            <FontAwesomeIcon icon={faUndo}/> Reset view
-          </button>
-          <br/><br/>
-          <button onClick={() => copyLink(routerLocation)}
-            className="action"
-            data-toggle="tooltip"
-            title="copy a link to this visualization to the clipboard">
-            <FontAwesomeIcon icon={faLink}/> Get link
-          </button>
+            {showDeGroupPicker &&
+            <>
+              <DeGroupPicker
+                exploreInfo={exploreInfo}
+                setShowDeGroupPicker={setShowDeGroupPicker}
+                setDeGenes={setDeGenes}
+                setDeGroup={setDeGroup}
+              />
+            </>
+            }
+            <button className="action"
+              onClick={clearExploreParams}
+              title="Reset all view options"
+              data-analytics-name="explore-view-options-reset">
+              <FontAwesomeIcon icon={faUndo}/> Reset view
+            </button>
+            <button onClick={() => copyLink(routerLocation)}
+              className="action"
+              data-toggle="tooltip"
+              title="Copy a link to this visualization to the clipboard">
+              <FontAwesomeIcon icon={faLink}/> Get link
+            </button>
+          </>
+          }
+          {deGenes &&
```

Comment:
Agreed.  I'll do this reorganizing while incorporating large user-facing changes in the MVP PR.

Reply:
Do we have a pressing need to get this into staging in the next day or so?  Merging suboptimal prototype code where it's confined to a prototype-specific file is usually ok.  But in core parts of SCP UX functionality, I don't want to get in the habit of merging "we'll clean it up later" code.  I'm happy to do the re-org myself if you don't want to lose velocity on the functionality improvements--my tickets this sprint are lower priority

Reply:
Yup, per the TODO in your other comment in this review, I planned to scope the high-level refactor into the PR for the next bolus of code and UX other refinements.  I think it'd be more efficient and negligible risk of perma-TODO, and suspect having another engineer do it would add substantial thrashing and communication overhead.

I hadn't appreciated from your original review that this refactor was a blocking concern for this PR.  I'll do it now!